watcher-c/readme.md

```diff
@@ -0,0 +1,34 @@
+# Using the watcher library from C
+
+The watcher can also be used from C by including the `watcher-c.h` file on compilation and in your project.
+
+You can cross compile the library via the `watcher-c/cross-compile.sh` script or download the precompiled library from the releases.
+
+### Compiling the library from source
+
+If you want to compile the library yourself you can follow the following example:
+
+First check out the `next` branch, compile a shared `libwatcher-c.so` library with `gcc` and install
+it in the `/usr/local/lib` directory:
+
+```bash
+curl -L https://github.com/e-dant/watcher/archive/refs/heads/next.tar.gz | tar xz
+cd watcher-next/watcher-c
+gcc -o libwatcher-c.so ./src/watcher-c.cpp -I ./include -I ../include -std=c++17 -O3 -Wall -Wextra -fPIC -shared
```

Comment:
I hesitate to add compiler optimizations in example instructions.

We can either be extremely aggressive in our optimizations (like some of our build systems do), or very example-focused and minimal (omitting O3, for example)

Same applies to Wall, Wextra -- for example commands, I think we can omit

Reply:
👍  I removed `Wall` `Wextra` `O3` and changed it to `c++`